ouroboros-network/src/Ouroboros/Network/BlockFetch/Client.hs

```diff
@@ -290,6 +294,14 @@ blockFetchClient _version controlMessageSTM
             -- Add the block to the chain DB, notifying of any new chains.
             addFetchedBlock (castPoint (blockPoint header)) block
 
+            let slotNo = case pointSlot $ blockPoint header  of
+                               Origin -> SlotNo 0 -- We don't care
+                               (At s) -> s
+            forgeTime_m <- atomically $ slotToTime slotNo
```

Comment:
I'd prefer to use`slotNo_m  = Maybe SlotNo` rather than `SlotNo 0 -- We don't care`, then
`join <$> traverse (atomically . slotToTime) slotNo_m` or something similar.